tool/tsh/tsh.go

```diff
@@ -1319,6 +1321,9 @@ func onLogin(cf *CLIConf) error {
 
 	key, err := tc.Login(cf.Context)
 	if err != nil {
+		if !usernameProvided && auth.IsInvalidLocalCredentialError(err) {
```

Comment:
nit: Isn't simpler to just check `cf.Username == ""` instead of using a boolean flag for that?

Reply:
`cf.Username` might have been set to `tc.Username` by this point, and `usernameProvided` is based on the original value. I actually just discovered that `CLIConf` has the `ExplicitUsername` field which does exactly the same thing, so I've removed `usernameProvided`.